### ISSUE TITLE
Add support for TypeScript

### ISSUE THREAD

**Issue by Shinigami92** [vivin/condicio#2]
Currently we can not use this package because it does not support TypeScript

Therefore, we would have to manually look up everything in the documentation, and this is not a very nice developer experience

![Bildschirmfoto 2019-08-22 um 10 36 54](https://user-images.githubusercontent.com/7195563/63499779-d8b8be00-c4c8-11e9-9e67-9d8fb4d612c4.png)

**Comment by vivin** [comment]
@Shinigami92  I completely agree. Unfortunately I don't know when I'll have time to look at this. I would like to convert this to TS at some point.
